plugins/LayerTree.jsx

```diff
@@ -787,6 +797,70 @@ class LayerTree extends React.Component {
         }, null, ' ');
         FileSaver.saveAs(new Blob([data], {type: "text/plain;charset=utf-8"}), layer.title + ".json");
     };
+
+    manageStyleSelector = (uuid) => {
+        const styleList = this.getLayerStyles(this.props.layers);
+        if (styleList.length > 0) {
+            this.setState((state) => ({
+                styleList: styleList,
+                activestyleselector: state.activestyleselector === uuid ? null : uuid
+            }));
+        }
+    };
+    getLayerStyles = (layerList) => {
+        const styleList = new Set();
+        const collectStyles = (layers) => {
+            for (const layer of layers) {
+                if (layer.sublayers && layer.sublayers.length > 0) {
+                    collectStyles(layer.sublayers);
+                }
+                if (layer.styles && Object.keys(layer.styles).length > 0) {
+                    for (const styleKey in layer.styles) {
+                        if (Object.prototype.hasOwnProperty.call(layer.styles, styleKey)) {
+                            styleList.add(layer.styles[styleKey]);
+                        }
+                    }
+                }
+            }
+        };
+        collectStyles(layerList);
+        return Array.from(styleList);
+    };
+    renderStyleSelector = () => {
```

Comment:
You could potentially re-use `renderStyleMenu` by modifying the `renderStyleMenu` arguments to also take the list of styles to render as well as the callback.